name.abuchen.portfolio/src/name/abuchen/portfolio/snapshot/filter/WithoutTaxesFilter.java

```diff
@@ -133,6 +133,9 @@ private void stripTaxes(PortfolioTransaction deliveryT, ReadOnlyPortfolio readOn
         else
             copy.setAmount(deliveryT.getAmount() + taxes.getAmount());
 
+        // copy units, except for taxes
+        deliveryT.getUnits().filter(u -> u.getType() != Unit.Type.TAX).forEach(u -> copy.addUnit(u));
```

Comment:
This is the corresponding (existing) code for buy/sell transactions:

https://github.com/buchen/portfolio/blob/master/name.abuchen.portfolio/src/name/abuchen/portfolio/snapshot/filter/WithoutTaxesFilter.java#L103-104